pkg/component/devfile/adapters/kubernetes/kubernetes.go

```diff
@@ -0,0 +1,68 @@
+package kubernetes
+
+import (
+	"github.com/golang/glog"
+	"github.com/openshift/odo/pkg/component/devfile/adapters/common"
+	devfileCommon "github.com/openshift/odo/pkg/devfile/versions/common"
+	"github.com/openshift/odo/pkg/kclient"
+	"github.com/openshift/odo/pkg/log"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+)
+
+// Adapter maps Devfiles to Kubernetes resources and actions
+type Adapter struct {
+	common.DevfileAdapter
+	Client kclient.Client
+}
+
+// Start creates Kubernetes resources that correspond to the devfile if they don't already exist
+func (k Adapter) Start() error {
+
+	err := k.Devfile.Data.Validate()
```

Comment:
Do we need to validation here? `devObj, err := devfile.Parse(pdo.devfilePath)` in `devfile.go` is already doing schema and data validation..

Reply:
Yes, should be parsed and validated before it's added to the context now so no need for it here anymore.